Add social recovery (#695)
w3f/polkadot-wiki: docs/kusama-social-recovery.md

---
id: kusama-social-recovery
title: Social Recovery
sidebar_label: Account Recovery
---

Managing an account is not an easy task. Many people have lost their private key due to improper key
management over the past few years. Kusama provides a method that allows users to recover their
accounts by setting up a social recovery. It is an M-of-N recovery tool that is based on the
multisignature wallet to get back access of your lost account.

> Note: There is no way to get back your private key by using this method. This is just a way of
> performing transactions on behalf of the lost account, so you can think of it like a proxy
> instead.

In this guide, you will learn how to create a recoverable account, how to recover it, and what you
need to be aware of when using it.

## Create a Recoverable Account

You will use your existing account to call `createRecovery` to select a list of accounts that you

> trust to help you recover the account when you lost the private key. To create a recoverable

Reviewer: ```suggestion
trust to help you recover the account when you lose the private key. To create a recoverable
```

account, you will be required to set a `threshold` that is the number of your friends who need to
approve the recovery process in order to recover your account.

![](/img/recovery/social-recovery-diag-1.png)

First, go to [PolkadotJS->Accounts](https://polkadot.js.org/apps/#/accounts) page that shows all

> available accounts on your browser's local storage. To create a recoverable account, making sure

Reviewer: ```suggestion
available accounts on your browser's local storage and PolkadotJS extension. To create a recoverable account, make sure
```

> that you have some KSMs to pay for the transaction fees and some will be used for reserved as well.

Reviewer: ```suggestion
that you have some KSMs to pay for the transaction fees. You will also need some for the reserve required by the account recovery setup.
```

I would also document how much it reserves so people know what to expect.

![](/img/recovery/social-recovery-1.png)

Then click the menu that is located besides the `send` button, and choose `Make recoverable`.

![](/img/recovery/social-recovery-2.png)

Now you need to provide the following information:

> `trusted social recovery helpers` - A list of accounts that you trust. These can help you if you
> lose the private key. Since setting up a recoverable account is required to lock up KSMs, better to
> ensure your account has enough transferable balance to cover it. As you select additional recovery
> helpers, more KSMs will be required to stay locked up.

Reviewer: ```suggestion
`trusted social recovery helpers` - A list of accounts that you trust. These can help you if you
lose the private key. Since setting up a recoverable account requires you to lock up KSM, ensure 
your account has enough transferable balance to cover it. As you select additional recovery
helpers, more KSM will be required.
```

> `recovery threshold` - The number of friends is required to submit a `vouchRecovery` transaction in

Reviewer: ```suggestion
`recovery threshold` - The number of friends required to submit a `vouchRecovery` transaction in
```

order to recover the account.

> Note: 1 is the minimum, but it is not recommended to set a small number. If you set 1, that means
> any of your recovery helpers would be able to recover your account.

`recovery block delay` - Once the threshold is reached, you will need to wait until the block delay
has passed until you can claim the recovery. This is a protection mechanism to allow the account

> owner to have enough time to check and react in case someone pretends to be you and does a recovery

Reviewer: ```suggestion
owner to have enough time to check and react in case someone pretends to be you and initiates a recovery
```

process.

> Note: Setting the block delay to be a little longer would be better since even if an attacker
> acquired enough signatures to recover your account, they would still have to wait until the block
> delay is passed in order to control your account.

![](/img/recovery/social-recovery-3.png)

## Recover your Account

This section would be showing you how to initiate a recovery process and get back the balance that

> held in the lost account to the new account.

Reviewer: Consider removing this paragraph

![](/img/recovery/social-recovery-diag-2.png)

The above diagram shows the whole process of recovering an account.

> > Note: Ensure that your new account has enough KSM for paying the transaction fees and the amount

Reviewer: ```suggestion
> Note: Ensure that your new account has enough KSM to pay for the transaction fees and the amount
```

> > that is used for reserved when doing the recovery.

Reviewer: ```suggestion
> that is used for reserve when doing the recovery.
```

Reviewer: I would also clarify how much these expenses might be

Navigate to the menu that is located besides the send button in the row of your new account and
click the "Initiate recover for another" option.

![](/img/recovery/social-recovery-4.png)

Then input the address you would like to recover in "recover this account" field and click "Start
recovery".

![](/img/recovery/social-recovery-5.png)

Once the transaction went through, some KSMs will be locked to prevent malicious behavior.

![](/img/recovery/social-recovery-6.png)

Now call your friends that you have set in the first section. They are required to submit a

> `vouchRecovery` transaction that provides the lost and the rescuer address.

Reviewer: ```suggestion
`vouchRecovery` transaction.
```

![](/img/recovery/social-recovery-7.png)

Once the threshold is reached and the block delay is passed, use the new account to submit a
`claimRecovery` transaction that would set a proxy on behalf of your lost account. It means that you
can still use the lost account in an indirect way to interact with the network.

![](/img/recovery/social-recovery-8.png)

To see the proxy information, use your new account by calling the "recovery->proxy(Accountid)"
function at the [Chain state](https://polkadot.js.org/apps/#/chainstate) page. It should point to
your lost account.

![](/img/recovery/social-recovery-9.png)

> Next, in order to call the "closeRecovery" transaction, you can make use of the "asRecovered" as

Reviewer: ```suggestion
Next, in order to call the "closeRecovery" transaction, you can make use of the "asRecovered" function as
```

your lost account to get the locked KSM.

![](/img/recovery/social-recovery-10.png)

> Once the transaction went through, the reserved KSM from the "NEW-ACC" has been moved to the lost

Reviewer: ```suggestion
Once the transaction goes through, the reserved KSM from the "NEW-ACC" will have been moved to the lost
```

> account. This is a way of preventing someone from recoversing other accounts maliciously. Imagine if

Reviewer: ```suggestion
account. This is a way of preventing someone from recovering other accounts maliciously. Imagine if
```

> someone tries to initiate recovery on your account, you can do this to slash their locked KSMs.

Reviewer: ```suggestion
someone tried to initiate recovery on your account, you can do this to slash their locked KSM.
```

![](/img/recovery/social-recovery-11.png)

> Moving on, by using the "asRecovered" to submit the `removeRecovery` transaction on behalf of the

Reviewer: ```suggestion
Moving on, we use the `asRecovered` function to submit the `removeRecovery` transaction on behalf of the
```

---
id: kusama-social-recovery
title: Social Recovery
sidebar_label: Account Recovery
---

Managing an account is not an easy task especially many people have lost their private key due to

> improper key management. Kusama provides a method that allows users to recover their accounts by

Reviewer 2: ```suggestion
Managing an account is not an easy task. Many people have lost their private key due to
improper key management. Kusama provides a method that allows users to recover their accounts by
```

setting up a social recovery. It is an M-of-N recovery tool that is based on the multisignature

> wallet to get back the control of your lost account.

Reviewer 2: ```suggestion
wallet to get back access of your lost account.
```

> Note: There is no way to get back your private key by using this method. This is just a way of
> performing transactions on behalf of the lost account, so you can think of it like a proxy
> instead.

In this guide, you will learn how to create a recoverable account, how to recover it, and what you
need to be aware of when using it.

## Create a Recoverable Account

You will use your existing account to call `createRecovery` to select a list of accounts that you
trust to help you recover the account when you lost the private key. To create a recoverable
account, you will be required to set a `threshold` that is the number of your friends who need to
approve the recovery process in order to recover your account.

![](/img/recovery/social-recovery-diag-1.png)

First, go to [PolkadotJS->Accounts](https://polkadot.js.org/apps/#/accounts) page that shows all
available accounts on your browser's local storage. To create a recoverable account, making sure
that you have some KSMs to pay for the transaction fees and some will be used for reserved as well.

![](/img/recovery/social-recovery-1.png)

> Then click the menu that besides the `send` button, and choose `Make recoverable`.

Reviewer 2: ```suggestion
Then click the menu that is located besides the `send` button, and choose `Make recoverable`.
```

![](/img/recovery/social-recovery-2.png)

Now you need to provide the following information:

> `trusted social recovery helpers` - A list of accounts that you trust can help you when you lost the

Reviewer 2: ```suggestion
`trusted social recovery helpers` - A list of accounts that you trust. These can help you if you lose the
```

private key. Since setting up a recoverable account is required to lock up KSMs, better to ensure

> sure your account has enough transferrable balance for this. As you select more recovery helpers,

Reviewer 2: ```suggestion
your account has enough transferable balance to cover it. As you select more recovery helpers,
```

> sure your account has enough transferrable balance for this. As you select more recovery helpers,
> more KSMs will be required to lock up.

Reviewer 2: ```suggestion
sure your account has enough transferrable balance for this. As you select additional recovery helpers,
more KSMs will be required to stay locked up.
```

`recovery threshold` - The number of friends is required to submit a `vouchRecovery` transaction in
order to recover the account.

> Note: 1 is the minimum, but it is not recommended to set a small number. If you set 1, that means
> any of your recovery helpers would be able to recover your account.

> `recovery block delay` -Once the threshold is reached, you will need to wait until the block delay

Reviewer 2: ```suggestion
`recovery block delay` - Once the threshold is reached, you will need to wait until the block delay
```

> is passed and then you can claim the recovery. This is a protection mechanism to allow the account

Reviewer 2: ```suggestion
has passed until you can claim the recovery. This is a protection mechanism to allow the account
```

> owner to have enough time to check and react in case someone pretends you to start the recovery

Reviewer 2: ```suggestion
owner to have enough time to check and react in case someone pretends to be you and does a recovery
```

process.

> > Note: set the block delay a little longer would be better since even they have acquired enough

Reviewer 2: ```suggestion
> Note: Setting the block delay to be a little longer would be better since even if an attacker acquired enough
```

> signatures to recover your account, they would still have to wait until the block delay is passed
> in order to control your account.

![](/img/recovery/social-recovery-3.png)

## Recover your Account

This section would be showing you how to initiate a recovery process and get back the balance that
held in the lost account to the new account.

![](/img/recovery/social-recovery-diag-2.png)

The above diagram shows the whole process of recovering an account.

> > Note: ensuring that your new account has enough KSM for paying the transaction fees and some are

Reviewer 2: ```suggestion
> Note: Ensure that your new account has enough KSM for paying the transaction fees and the amount that is
```

> used for reserved when doing the recovery.

First, use your new account that shows in the [Account](https://polkadot.js.org/apps/#/accounts)
page to initiate a recovery process by clicking the "Initiate recovery for another" option that is

> in the menu besides the send button.

Reviewer 2: "Navigate to the menu that is located besides the send button in the row of your new account and click the "Initiate recover for another" option.

![](/img/recovery/social-recovery-4.png)

Then input the address you would like to recover in "recover this account" field and click "Start
recovery".

![](/img/recovery/social-recovery-5.png)

Once the transaction went through, some KSMs will be locked to prevent malicious behavior.

![](/img/recovery/social-recovery-6.png)

Now call your friends that you have set in the first section. They are required to submit a
`vouchRecovery` transaction that provides the lost and the rescuer address.

![](/img/recovery/social-recovery-7.png)

Once the threshold is reached and the block delay is passed, use the new account to submit a

> `claimRecovery` transaction that would set a proxy on behalf of your lost account in the underlying.

Reviewer 2: ```suggestion
`claimRecovery` transaction that would set a proxy on behalf of your lost account.
```

It means that you can still use the lost account in an indirect way to interact with the network.

![](/img/recovery/social-recovery-8.png)

To see the proxy information, use your new account by calling the "recovery->proxy(Accountid)"
function at the [Chain state](https://polkadot.js.org/apps/#/chainstate) page. It should point to
your lost account.

![](/img/recovery/social-recovery-9.png)

Next, in order to call the "closeRecovery" transaction, you can make use of the "asRecovered" as
your lost account to get the locked KSM.

![](/img/recovery/social-recovery-10.png)

Once the transaction went through, the reserved KSM from the "NEW-ACC" has been moved to the lost

> account. This is a way of preventing someone from recovers other accounts maliciously. Imagine if

Reviewer 2: ```suggestion
account. This is a way of preventing someone from recoversing other accounts maliciously. Imagine if
```

someone tries to initiate recovery on your account, you can do this to slash their locked KSMs.

![](/img/recovery/social-recovery-11.png)

Moving on, by using the "asRecovered" to submit the `removeRecovery` transaction on behalf of the
lost account to release the reserved KSM from your lost account.

![](/img/recovery/social-recovery-12.png)

Now all your account balance should be transferable.

![](/img/recovery/social-recovery-13.png)

Finally, transfer all of your available balance from the lost account to the new account.

![](/img/recovery/social-recovery-14.png)

Congratulation! You have successfully done the recovery process.

![](/img/recovery/social-recovery-15.png)

There is still one possible way to recover the account without going through the recovery process.

> That is root. However, while using the root permission to perform this action, you will have to go

Reviewer 2: ```suggestion
That is by using the `Root` origin. However, while using the root permission to perform this action, you will have to go
```

through the governance by convincing the council members to help you to submit a council proposal or

> submit a public proposal yourself to see whether they will approve it or not. To learn more about

Reviewer 2: Rephrase ->

In order to use root permissions you will need to either go through the council or submit a public proposal.
